### PR TITLE
[Register Agent] Add remember server address check

### LINKED ISSUE #4718: Add an option to easily save the Wazuh server address in the "Deploy new agent" section
**Describe the solution you'd like**

An all-in-one deployment, in which all the Wazuh central components are installed on the same server, is very common. For example, the [OVA](https://documentation.wazuh.com/current/deployment-options/virtual-machine/virtual-machine.html) or the installation resulting from following the [Quickstart](https://documentation.wazuh.com/current/quickstart.html) section. 

In the **Agents** > **Deploy new agent** section, we find that the suggested Wazuh server address is `localhost` for this type of deployment. 

This `localhost` suggestion is useless, and the user has to enter the server address every time he wants to deploy a new agent. 

To modify the default value, the user has to go to **Settings** > **Configuration** and edit the **Enrollment DNS** value.  This process is not trivial nor known by many users. 

My suggestion is to add a button to easily save the Wazuh server address. See the image below. 

image

### PR DESCRIPTION
**Pull request by Machi3mfl**
### Description

Added toggle `Remember server address` button to save `enrollment.dns` configuration when is checked.

### Issues Resolved
Closes #4718

### Evidence

image

### 1. When the user checks the `Remember server address` button this will save the value on the configuration

image

image

### 2. When the server address input is `not valid` the toggle must be `disabled`

https://github.com/wazuh/wazuh-kibana-app/assets/6089438/b2efc52b-ec60-47c4-a642-22455c4f0bca

###  3. When the server address is equal to the value saved on the configuration `enrollment.dns` setting the toggle must be checked

image

image

### 4. When the server address value is valid and is not equal to the value saved in Configuration the toggle must be false.

https://github.com/wazuh/wazuh-kibana-app/assets/6089438/24a0823d-379a-431c-8330-f2706d9b4203

### Check List
- [x] All tests pass
  - [x] `yarn test:jest`
- [x] New functionality includes testing.
- [x] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by chantal-kelm — APPROVED** [review]
## **With unsaved IP:**

Captura de pantalla 2023-08-17 a la(s) 15 45 25

## **With saved IP:**

Captura de pantalla 2023-08-17 a la(s) 15 46 44
Captura de pantalla 2023-08-17 a la(s) 15 47 09

File: plugins/main/public/controllers/register-agent/components/server-address/server-address.tsx [hunk 1]
```diff
@@ -42,6 +46,50 @@ const ServerAddressInput = (props: ServerAddressInputProps) => {
       isPopoverServerAddress => !isPopoverServerAddress,
     );
   const closeServerAddress = () => setIsPopoverServerAddress(false);
+  const [rememberServerAddress, setRememberServerAddress] = useState(false);
+  const [defaultServerAddress, setDefaultServerAddress] = useState(formField.initialValue ? formField.initialValue : '')
```
Comment (jbiset) [inline plugins/main/public/controllers/register-agent/components/server-address/server-address.tsx:50]: `formField` can be `undefined`
Idem in other lines

**Review by jbiset — CHANGES_REQUESTED** [review]
CR :yellow_circle:    (Request changes)

## TEST UI   :green_circle:  
### 1. :green_circle:  When the user checks the Remember server address button this will save the value on the configuration

Evidence_1_remember_server_address_A

Evidence_1_remember_server_address_B

### 2. :green_circle:  When the server address input is not valid the toggle must be disabled

Evidence_2_invalid_IP_toggle_disabled_A

Evidence_2_invalid_IP_toggle_disabled_B

Evidence_2_invalid_IP_toggle_disabled_C

### 3. :green_circle:  When the server address is equal to the value saved on the configuration enrollment.dns setting the toggle must be checked

Evidence_3_valid_IP_saved_toggle_enabled

### 4. :green_circle:  When the server address value is valid and is not equal to the value saved in Configuration the toggle must be false.

Evidence_4_valid_IP_NOT_saved_toggle_disabled

File: plugins/main/public/controllers/register-agent/components/server-address/server-address.tsx (hunk 1 above)
Comment (Machi3mfl) [reply plugins/main/public/controllers/register-agent/components/server-address/server-address.tsx]: Fixed in [9d04786](https://github.com/wazuh/wazuh-kibana-app/pull/5791/commits/9d047862f9dc464d091a118427379b1994a45478)

**Review by jbiset — APPROVED** [review]

**Pull request merged**